core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/SampleSheet.java

```diff
@@ -274,34 +274,33 @@ public String createSampleSheet(Run run, User user) {
       return Stream.of(makeColumns(false, model, partitionNumber, partitionBarcode, aliquot, Collections.emptyList(), userName)
           .collect(Collectors.joining(",")));
     }
-    List<List<String>> indices = null;
     Index index1 = aliquot.getParentLibrary().getIndex1();
     Index index2 = aliquot.getParentLibrary().getIndex2();
-    if (index1.getFamily().hasFakeSequence()) {
-      indices = index1.getRealSequences().stream()
+    Set<String> sequence1s = getSequences(index1, model);
+    List<List<String>> indices = null;
+    if (index2 == null) {
+      indices = sequence1s.stream()
           .map(Collections::singletonList)
           .collect(Collectors.toList());
     } else {
-      indices = Arrays.asList(Collections.singletonList(index1.getSequence()));
-    }
-    if (index2 != null) {
-      if (index2.getFamily().hasFakeSequence()) {
-        indices.addAll(index2.getRealSequences().stream()
-            .map(model.getDataManglingPolicy() == InstrumentDataManglingPolicy.I5_RC ? SampleSheet::reverseComplement : Function.identity())
-            .map(Collections::singletonList)
-            .collect(Collectors.toList()));
-      } else {
-        String sequence = model.getDataManglingPolicy() == InstrumentDataManglingPolicy.I5_RC
-            ? SampleSheet.reverseComplement(index2.getSequence())
-            : index2.getSequence();
-        indices.add(Collections.singletonList(sequence));
-      }
+      Set<String> sequence2s = getSequences(index2, model);
+      indices = sequence1s.stream()
+          .flatMap(sequence1 -> sequence2s.stream()
+              .map(sequence2 -> Arrays.asList(sequence1, sequence2)))
+          .collect(Collectors.toList());
     }
     final boolean needsSuffix = indices.size() > 1;
     return indices.stream()
         .map(index -> makeColumns(needsSuffix, model, partitionNumber, partitionBarcode, aliquot, index, userName)
             .collect(Collectors.joining(",")));
+  }
 
+  private Set<String> getSequences(Index index, InstrumentModel model) {
+    Set<String> sequences = index.getFamily().hasFakeSequence() ? index.getRealSequences() : Collections.singleton(index.getSequence());
+    if (model.getDataManglingPolicy() == InstrumentDataManglingPolicy.I5_RC) {
```

Comment:
I think this also needs to check if the index is in position 2 since the position 1 shouldn't be reverse complemented.